packages/d3fc-webgl/README.md

```diff
@@ -1,16 +1,750 @@
 # d3fc-webgl
 
-A collection of WebGL shaders and supporting components used by [d3fc-series](https://github.com/d3fc/d3fc/tree/master/packages/d3fc-series#d3fc-series) to render WebGL series.
+A collection of very low-level components for rendering with WebGL. See other packages (e.g. [d3fc-series](https://github.com/d3fc/d3fc/tree/master/packages/d3fc-series#d3fc-series)) for higher-level components which make use of these low-level components.
 
-This package is still a work in progress and its API has not been finalised. Therefore, it is recommended that this package is not used directly.
+[Main D3FC package](https://github.com/d3fc/d3fc)
 
-# Shader Naming Convention
+## Installing
+
+```bash
+npm install @d3fc/d3fc-webgl
+```
+
+## API Reference
+
+* [Series](#series)
+  * [Common Properties](#common-properties)
+  * [Area](#area)
+  * [Bar](#bar)
+  * [BoxPlot](#boxplot)
+  * [Candlestick](#candlestick)
+  * [ErrorBar](#errorbar)
+  * [Line](#line)
+  * [Ohlc](#ohlc)
+  * [Point](#point)
+* [Buffers](#buffers)
+  * [Attribute Buffer Builders](#attribute-buffer-builders)
+    * [Element Attribute](#element-attribute)
+    * [Adjacent Element Attribute](#adjacent-element-attribute)
+    * [Vertex Attribute](#vertex-attribute)
+    * [Base Attribute](#base-attribute)
+  * [Uniform Builder](#uniform-builder)
+  * [Buffer Builder](#buffer-builder)
+  * [Element Indices](#element-indices)
+  * [Types](#types)
+* [Shader Naming Convention](#shader-naming-convention)
+
+This package contains the components needed to render a standard or custom series with WebGL. The standard series share a common API with a typical configuration requiring x and y WebGL scales together with a number of attribute buffers.
+
+### Series
+
+The series components can be used to generate a WebGL series of the relevant chart, all chart types are supported.
+
+Some of the series make use of `next`/`previous` attributes. These are needed in situations where the current element requires information about adjacent elements in order to be drawn correctly. For example when drawing a line we need to know information about adjacent data points so that we can draw the connection between them correctly.
+
+Currently only the [line](#line) and [point](#point) series are able to render different orientations (`horizontal` and `vertical`), all other charts render as a `vertical` chart. Despite this the series components maintain an agnostic naming convention using `cross` and `main` values to provide a consistent interface and to better support orientation in the future.
+
+Most values used in the series components are scaled during the execution of the vertex shader, the scaling values are provided by the [WebGL scales](link to scales). The exception to this rule are values provided as properties rather than data point values, for example `bandwidth` and `point size`. Instead these properties should be given values in an appropriate scale as documented.
```

Comment:
Sorry I know I asked for this paragraph (`and how the values are scaled`) but I can't remember what I was referring to. As it is, I'm not actually sure it's needed.